IDE project configuration for developing Godot GDExtensions using Rider

van800 opened the pull request:
If you are interested in this addition, please let me know.
I would test more specifically with your improved template.
Currently I expect, rename project would not be automatically propagated to vcxproj.

There is a short demo in the https://github.com/godotengine/godot-cpp-template/pull/95 comments.

nikoladevelops commented:
Also I should let you know that currently debug builds are built without debug symbols by default, maybe this should also be included in that option, since you won't really be able to attach a debugger without debug symbols.

Also the debug symbols slow down your build (at least on windows) so this option should be switched/toggled whenever the user decides.

There's a lot to take in, so when you have time play around with all this and experiment

nikoladevelops commented:
As soon as you have a working solution I'll test the branch and see if it's good.

I still believe that if the project supports Rider by adding additional files, it should also support CLion and other commonly used editors (Zed would be very good since it's completely free to use forever even for closed source or commercial projects)

If you can do that it would be nice, but if you think that it's out of scope for this pr I get it.

van800 commented:
The sln+vcxproj configuration files are pretty much only Rider and Visual Studio. So I can only ensure compatibility with those two in the scope of this pull request.

nikoladevelops commented:
Also all of this should be compatible with GitHub actions no matter which editor is used, the source files should still end up in the src directory (check the workflow I have right now for building it for every platform)

van800 commented:
I am sorry, I am under pressure on different tracks lately, so no capacity to continue for now, but I hope to return to it later.
Please feel free to close the PR, or I would prefer it as a draft.

Jean-H77 committed:
Update README.md

`libgdexample` string has been replaced by `EXTENSION-NAME` in `demo/bin/example.gdextension

paddy-exe committed:
Merge pull request #96 from Jean-H77/patch-1

van800 committed:
fix cmake compilation of the example project

paddy-exe committed:
Merge pull request #97 from van800/cmake2

Chkoupinator committed:
Add subdirectories to source file collection logic in SConstruct up to 3 layers of depth

Added automatic file collection using Glob so that source files in directories that are up to 3 layers of depth into src/ get properly detected by SConstruct.

Ivorforce committed:
Enhance README usage instructions, and add links

Ivorforce committed:
Split CI from build actions, since they're for pretty different use-cases.
Reduce CI build combinations to an explicit 5.

paddy-exe committed:
Merge pull request #98 from Chkoupinator/patch-1

paddy-exe committed:
Merge pull request #100 from Ivorforce/usage-update

Enhance README usage instructions, and add links

Ivorforce committed:
Revert "Add subdirectories to source file collection logic in SConstruct up to 3 layers of depth"

This reverts commit 84c86b3d1457b00aa0dbb94879e9bf3beeb25c8e.

paddy-exe committed:
Merge pull request #101 from Ivorforce/ci-cd-builds

Split CI from build actions, since they're for pretty different use-cases

MOS8367R0 committed:
Fix macos.double.release and web.wasm32.double.debug paths

paddy-exe committed:
Merge pull request #103 from MOS8367R0/main

paddy-exe committed:
Rename demo folder to project for generalizing the template usecase

Renames all references to the demo folder and changes it to "project"

paddy-exe committed:
Merge pull request #109 from paddy-exe/rename-demo-folder-to-project-folder

Rename demo folder to project for generalizing the template usecase

paddy-exe committed:
Add reloadable flag to gdextension file

adds the reloadable flag to the .gdextension file for quickly getting
started and having the extension be reloadable almost from the get-go

paddy-exe committed:
Merge pull request #110 from paddy-exe/add-reloadable-gdextension-file

Add reloadable flag to gdextension file

van800 committed:
remove caching of the LIBNAME in CMake configuration to let the generated run configuration in Rider/CLion to updated as soon as the library name is changed, fixes https://youtrack.jetbrains.com/issue/RIDER-136871

Ivorforce committed:
Merge pull request #113 from van800/libname

remove caching of the LIBNAME in CMake configuration

van800 committed:
Create the godot-cpp/gen/include directory at configure time so IDEs can see generated headers immediately after generate_bindings target produces them
